packages/react-components/react-color-picker-preview/library/src/components/ColorArea/useColorArea.ts

```diff
@@ -12,20 +21,199 @@ import type { ColorAreaProps, ColorAreaState } from './ColorArea.types';
  * @param ref - reference to root HTMLDivElement of ColorArea
  */
 export const useColorArea_unstable = (props: ColorAreaProps, ref: React.Ref<HTMLDivElement>): ColorAreaState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { targetDocument } = useFluent();
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const xRef = React.useRef<HTMLInputElement>(null);
+  const yRef = React.useRef<HTMLInputElement>(null);
+  const focusWithinRef = useFocusWithin();
+
+  const {
+    onChange,
+
+    // Slots
+    inputX,
+    inputY,
+    thumb,
+  } = props;
+
+  const [color, setColor] = useControllableState<HsvColor>({
+    defaultState: props.defaultColor,
+    state: props.color,
+    initialState: INITIAL_COLOR_HSV,
+  });
+  const saturation = Math.round(color.s * 100);
+  const value = Math.round(color.v * 100);
+
+  const [activeAxis, setActiveAxis] = React.useState<'x' | 'y'>('x');
+
+  const requestColorChange = useEventCallback((event: MouseEvent) => {
+    if (!rootRef.current) {
+      return;
+    }
+
+    const coordinates = getCoordinates(rootRef.current, event);
+    const newColor: HsvColor = {
+      ...color,
+      s: coordinates.x,
+      v: coordinates.y,
+    };
+
+    setColor(newColor);
+    onChange?.(event, { type: 'change', event, color: newColor });
+  });
+
+  const handleDocumentMouseMove = React.useCallback(
+    (event: MouseEvent) => {
+      requestColorChange(event);
+    },
+    [requestColorChange],
+  );
+  const handleDocumentMouseUp = useEventCallback(() => {
+    targetDocument?.removeEventListener('mousemove', handleDocumentMouseMove);
+  });
+
+  const handleRootOnMouseDown: React.MouseEventHandler<HTMLDivElement> = useEventCallback(event => {
+    event.stopPropagation();
+    event.preventDefault();
+
+    requestColorChange(event.nativeEvent);
+
+    targetDocument?.addEventListener('mousemove', handleDocumentMouseMove);
+    targetDocument?.addEventListener('mouseup', handleDocumentMouseUp, { once: true });
+  });
+
+  const handleInputOnChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(event => {
+    const value = Number(event.target.value) / 100;
+    const newColor: HsvColor = {
+      ...color,
+      ...(event.target === xRef.current && { s: value }),
+      ...(event.target === yRef.current && { v: value }),
+    };
+
+    setColor(newColor);
+    onChange?.(event, { type: 'change', event, color: newColor });
+  });
+
+  const handleRootOnKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
`root` is `div` element:

```suggestion
  const handleRootOnKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
```